File: mad3004-introd-swift/C0711561_F2017_MAD3004_FinalProject/C0711561_F2017_MAD3004_FinalProject/FullTime.swift
import Foundation

public class FullTime : Employee {
    let typeOfEmployee = "FullTime"
    
    private var _salary: Double
    public var salary: Double {
        get {
            return _salary
        }
        
        set {
            _salary = newValue
        }
    }
    
    private var _bonus: Double
    public var bonus: Double {
        get {
            return _bonus
        }
        
        set {
            _bonus = newValue
        }
    }
    
    override init() {
        _salary = 0
        _bonus = 0
        super.init();
    }
    
    init(ppName: String, ppAge: Int, pSalary: Double, pBonus: Double) {
        _salary = pSalary
        _bonus = pBonus
        super.init(ppName, ppAge)
    }
    
    init(ppName: String, ppAge: Int, pSalary: Double, pBonus: Double, ppV: Vehicle) {
        _salary = pSalary
        _bonus = pBonus
        super.init(ppName, ppAge, ppV)
    }
    
    override func calcEarnings() -> Double {
        return Double(salary + bonus)
    }
    
    override func getTypeOfEmployee() -> String {
        return typeOfEmployee
    }
    
    override func printMyData() -> String {
        return "\(super.printMyData())\n  -Salary: \(Util.toCurrencyFormatFrom(value: salary))\n  -Bonus: \(Util.toCurrencyFormatFrom(value: bonus))\n  -Earnings: \(Util.toCurrencyFormatFrom(value: calcEarnings()))"
    }
}

File: mad3464-programming-java-i/C0711561_F2017_MAD3464_FinalProject/src/c0711561/f2017/mad3464/finalproject/employee/FullTime.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package c0711561.f2017.mad3464.finalproject.employee;

import c0711561.f2017.mad3464.finalproject.Util;
import c0711561.f2017.mad3464.finalproject.vehicle.Vehicle;
import java.math.BigDecimal;

/**
 *
 * @author macstudent
 */
public class FullTime extends Employee {
    
    private final double salary;
    private final double bonus;

    public FullTime(String name, int age, double salary, double bonus) {
        super(name, age);
        this.salary = salary;
        this.bonus = bonus;
    }
    
        public FullTime(String name, int age, double salary, double bonus, Vehicle vehicle) {
        super(name, age, vehicle);
        this.salary = salary;
        this.bonus = bonus;
    }

    @Override
    public BigDecimal calcEarnings() {
        return new BigDecimal(salary + bonus);
    }

    @Override
    public String getTypeOfEmployee() {
        return "Full Time";
    }

    @Override
    public String printMyData() {
        String data = super.printMyData() 
                + "\n -Salary: " + Util.toCurrencyFormatFrom(salary) 
                + "\n -Bonus: " + Util.toCurrencyFormatFrom(bonus) 
                + "\n -Earnings: " + Util.toCurrencyFormatFrom(calcEarnings());
        
        return data;
    }
    
    
    
}
